packages/client/src/views/ReviewCorrection/ReviewCorrection.tsx

```diff
@@ -622,11 +622,10 @@ function applyCorrectionToData(record: IDeclaration) {
 
       switch (typeof record.data[curr.valueCode]?.[curr.valueId]) {
         case 'number':
-          acc[curr.valueCode][curr.valueId] = Number(curr.valueString)
+          acc[curr.valueCode][curr.valueId] = curr.newValueNumber
           break
-
         default:
-          acc[curr.valueCode][curr.valueId] = curr.valueString
+          acc[curr.valueCode][curr.valueId] = curr.newValueString
           break
       }
```

Comment:
Instead of testing the value of the field with 

`typeof record.data[curr.valueCode]?.[curr.valueId]` you can do 

```ts
switch (true) {
  case curr.valueString !== undefined:
    acc[curr.valueCode][curr.valueId] = curr.valueString
    break
  case curr.valueInteger  !== undefined
    ...
}
```